### PR TITLE
DEV-5602 & DEV-5629 COVID recipient search bar and links update

### PR DESCRIPTION
**Pull request by ebdabbs**
**High level description:**
Adds a search bar to the COVID Profile **Spending by Recipient** section > **Recipients** tab table

**Technical details:**

- Adds `query` param to API request when the search string is more than 3 characters
- Creates a `SearchBar` component that could be reused in other sections of the page
- Styles substring of the recipient name that matches the search input
- When only one recipient profile page is available, link the recipient name instead of using `as Child`/ `as Recipient`
- Fixes a bug causing the Loans and Direct Payment tab counts to be swapped

**JIRA Ticket:**
[DEV-5602](https://federal-spending-transparency.atlassian.net/browse/DEV-5602) (search bar), [DEV-5629](https://federal-spending-transparency.atlassian.net/browse/DEV-5629) (recipient links)

**Mockup:**
https://bahdigital.invisionapp.com/share/TPIAEAE7HVG#/296042274_COVID-_Recipients_-_Recipient_Location_-_Recipient

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA tickets
- [x] Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [x] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [x] Added Unit Tests for methods in Container Components, reducers, helper functions, and models `if applicable`

Reviewer(s):
- [x] Design review complete @AGuyNamedMarco & @marcyheld 
- [ ] Code review complete

### CODE REVIEW

File: src/js/components/covid19/SearchBar.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+const SearchBar = ({ setQuery, query }) => {
+    const [searchString, setSearchString] = useState(query);
+
+    const onChange = (e) => {
+        setSearchString(e.target.value);
+    };
+
+    const onSubmit = (e) => {
+        e.preventDefault();
```
Comment (jonhill13) [inline src/js/components/covid19/SearchBar.jsx:23]: Any reason we are passing the event here?

File: src/js/components/covid19/SearchBar.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        setQuery(searchString);
+    };
+
+    const resetSearch = (e) => {
```
Comment (jonhill13) [inline src/js/components/covid19/SearchBar.jsx:27]: Any reason we are passing the event here?

File: src/js/components/covid19/SearchBar.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        e.preventDefault();
+        setSearchString('');
+        setQuery('');
+    };
+
+    const handleClick = (e) => {
+        if (searchString && query === searchString) {
+            resetSearch(e);
```
Comment (jonhill13) [inline src/js/components/covid19/SearchBar.jsx:35]: Any reason we are passing the event here?

File: src/js/components/covid19/SearchBar.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        }
+        else {
+            onSubmit(e);
```
Comment (jonhill13) [inline src/js/components/covid19/SearchBar.jsx:38]: Any reason we are passing the event here?

File: src/js/components/covid19/SearchBar.jsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+const propTypes = {
+    setQuery: PropTypes.func.isRequired,
+    query: PropTypes.string
+};
+
+const SearchBar = ({ setQuery, query }) => {
```
Comment (jonhill13) [inline src/js/components/covid19/SearchBar.jsx:15]: Any reason we are passing query here?

Comment (ebdabbs) [reply src/js/components/covid19/SearchBar.jsx:15]: `query` (what the table is currently filtered on) is compared to `searchString` (value of the input) to determine whether `handleClick` will reset or perform a new search (and to display the corresponding icon)

Comment (jonhill13) [reply src/js/components/covid19/SearchBar.jsx:15]: Yes, I guess I am confused? The search bar should be taking input and searching. I do not understand why search bar needs a parent to pass it the search term? There must be a case you are trying to handle here, I do not think I see it ( possibly when a user types the same input again? ).

Comment (ebdabbs) [reply src/js/components/covid19/SearchBar.jsx:15]: Right, if their input matches the existing search it prevents submitting the same search term again and instead gives them the option to reset.

Comment (jonhill13) [reply src/js/components/covid19/SearchBar.jsx:15]: Yeah, it just seems like this component should be self sufficient. The search bar should be able to handle that logic itself.  We could remove the query param and access the input element onSubmit to get the value or move that query parameter to the search bar like

```const SearchBar = ({ setQuery }) => {
    const [searchString, setSearchString] = useState('');
    const [currentSearchTerm, setCurrentSearchTerm] = useState('');

    const onChange = (e) => {
        setSearchString(e.target.value);
    };

    const onSubmit = (e) => {
        e.preventDefault();
        setCurrentSearchTerm(searchString);
        setQuery(searchString);
    };

    const resetSearch = (e) => {
        e.preventDefault();
        setSearchString('');
        setQuery('');
    };

    const handleClick = (e) => {
        if (searchString && currentSearchTerm === searchString) {
            resetSearch(e);
        }
        else {
            onSubmit(e);
        }
    };```

Thus, in the future when other devs implement this I need to do less work by just passing setQuery

`<SearchBar setQuery={search} />`

and not worry about handling the query param.

**Pull request merged**